vpr/src/place/move_transactions.cpp

```diff
@@ -33,6 +62,20 @@ e_block_move_result record_block_move(t_pl_blocks_to_be_moved& blocks_affected,
     return e_block_move_result::VALID;
 }
 
+//Moves the blocks in blocks_affected to their new locations
+void apply_move_blocks(const t_pl_atom_blocks_to_be_moved& blocks_affected) {
+    const auto& atom_lookup = g_vpr_ctx.atom().lookup;
+    std::set<ClusterBlockId> seen_clusters;
+    for (int blk_idx = 0; blk_idx < blocks_affected.num_moved_blocks; blk_idx++) {
```

Comment:
For performance, it may be a good idea to move the upper bound out of the for loop. This adds another line of code, but if this loop iterates a lot of times it may save some performance.

Reply:
You mean for each iteration it would go and read the value from the memory?

Reply:
By the syntax of the C++ for loops, the upper bound MUST be recalculated each iteration. For this example, this is because the compiler is not sure if you are changing the size of the blocks_affected array during the iteration (for example if your loop added elements to the end of the blocks_affected). The compiler needs to ensure that the it matches this functionality so it will read the value from memory each iteration, which *may* effect how well the compiler can perform optimizations since the upper bound of the loop may change each iteration.